XRTK-Core/Packages/com.xrtk.core/Providers/Controllers/BaseController.cs

```diff
@@ -244,21 +240,36 @@ internal async Task TryRenderControllerModelAsync(Type controllerType, byte[] gl
                 }
             }
 
-            // If we've got a controller model prefab, then place it in the scene.
+            // If we've got a controller model, then place it in the scene and get/attach the visualizer.
             if (controllerModel != null)
             {
-                var controllerObject = UnityEngine.Object.Instantiate(controllerModel, MixedRealityToolkit.Instance.MixedRealityPlayspace);
-                controllerObject.name = $"{controllerType.Name}_{controllerObject.name}";
-                Visualizer = controllerObject.GetComponent<IMixedRealityControllerVisualizer>();
+                //If the model was loaded from a system template
+                if (useSystemDefaultModels && gltfObject != null)
+                {
+                    controllerModel.name = $"{controllerType.Name}_Visualization";
```

Comment:
Could it be that controllerType gets null in here?

Reply:
It's expected all the way through this function that you might never get a model, so yes, the end result could be null.  Hence all the fallbacks to an eventual catch-all "no model" condition